tilequeue/rawr.py

```diff
@@ -377,54 +402,59 @@ def __call__(self):
                 self.log_exception(e, 'unmarshall payload')
                 continue
 
-            try:
-                parent = common_parent(coords, self.group_by_zoom)
-            except Exception as e:
-                self.log_exception(e, 'find parent')
-                continue
-
-            try:
-                rawr_tile_coord = convert_coord_object(parent)
-            except Exception as e:
-                self.log_exception(e, 'convert coord', parent)
-                continue
-
-            try:
-                rawr_gen_timing = {}
-                with time_block(rawr_gen_timing, 'total'):
-                    # grab connection
-                    with self.conn_ctx() as conn:
-                        # commit transaction
-                        with conn as conn:
-                            # cleanup cursor resources
-                            with conn.cursor() as cur:
-                                table_reader = TableReader(cur)
-
-                                rawr_gen_specific_timing = self.rawr_gen(
-                                    table_reader, rawr_tile_coord)
-
-                rawr_gen_timing.update(rawr_gen_specific_timing)
-                timing['rawr_gen'] = rawr_gen_timing
+            # split coordinates into group by zoom and higher and low zoom
+            # the message payload is either coordinates that are at group by
+            # zoom and higher, or all below the group by zoom
+            is_low_zoom = False
+            did_rawr_tile_gen = False
+            for coord in coords:
+                if coord.zoom < self.group_by_zoom:
+                    is_low_zoom = True
+                else:
+                    assert not is_low_zoom, \
+                        'Mix of low/high zoom coords in payload'
+
+            # check if we need to generate the rawr tile
+            # proceed directly to enqueueing the coordinates if not
+            if not is_low_zoom:
+                did_rawr_tile_gen = True
+                try:
+                    parent = common_parent(coords, self.group_by_zoom)
```

Comment:
I think we should also do:

```python
assert parent.zoom >= self.group_by_zoom
parent = parent.zoomTo(self.group_by_zoom).container()
```

I feel like there are going to be places in the code where we assume parent is a RAWR tile coordinate, but we'll only get a job for the partial tree and end up with `parent.zoom=11` or something like that. What do you think?

Reply:
Ooops... I should have looked at the definition of `common_parent`. That's exactly what it does already.